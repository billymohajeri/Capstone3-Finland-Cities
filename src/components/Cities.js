import { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink } from 'react-router-dom';
import fetchCities from '../redux/citiesApi';
import '../styles/Cities.css';

const Cities = () => {
  const state = useSelector((state) => state.cities);
  const dispatch = useDispatch();
  const [search, setSearch] = useState('');

  const handleChange = (e) => {
    setSearch(e.target.value);
  };

  const filtered = state.filter((item) => item.name.toLowerCase().includes(search.toLowerCase()));

  useEffect(() => {
    dispatch(fetchCities());
  }, [dispatch]);

  return (
    <>
      <div className="cities-search">
        <form className="search-form">
          <input
            className="search-input"
            type="text"
            value={search}
            aria-label="search"
            onChange={handleChange}
            placeholder="Search a city..."
          />
          <p className="message">
            {filtered.length}
            &nbsp;Items found...
          </p>
        </form>
      </div>
      <div className="cities-container">
        {filtered.map((city) => (
          <div key={city.id} className="city">
            <NavLink to={`/${city.id}`} className="city-link">
              <img src={`https://7gg9.short.gy/COA${city.id}`} alt="coat-of-arms" />
              <h2>{city.name}</h2>
            </NavLink>
          </div>
        ))}
      </div>
    </>
  );
};

export default Cities;
