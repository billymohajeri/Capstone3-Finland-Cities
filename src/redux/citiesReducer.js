const GET_CITIES = 'CITY/GET_CITIES';

const initialState = [];

const citiesReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_CITIES:
      return action.payload;
    default:
      return state;
  }
};

export const getCities = (payload) => ({
  type: GET_CITIES,
  payload,
});

export default citiesReducer;
