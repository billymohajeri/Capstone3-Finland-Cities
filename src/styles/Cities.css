.app-title,
.search-form {
  text-align: center;
}

.search-input {
  width: 50%;
  max-width: 500px;
}

img {
  -webkit-filter: saturate(0%);
  filter: saturate(0%);
  opacity: 40%;
  height: 200px;
}

img:hover {
  -webkit-filter: unset;
  filter: unset;

  /* filter: saturate(10%); */
  opacity: unset;
  transition: 2s;
}
