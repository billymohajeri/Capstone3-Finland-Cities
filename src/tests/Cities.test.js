import React from 'react';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import Cities from '../components/Cities';
import citiesReducer from '../redux/citiesReducer';

const createTestStore = () => {
  const store = configureStore({
    reducer: {
      cities: citiesReducer,
    },
  });

  return store;
};
export default createTestStore;

describe('Connected React/Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = createTestStore();

    component = renderer
      .create(
        <Provider store={store}>
          <Cities />
        </Provider>,
      )
      .toJSON();
  });

  test('Renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
