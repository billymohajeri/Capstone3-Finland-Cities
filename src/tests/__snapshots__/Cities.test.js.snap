// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected React/Redux Component Renders correctly 1`] = `
Array [
  <div
    className="cities-search"
  >
    <form
      className="search-form"
    >
      <input
        aria-label="search"
        className="search-input"
        onChange={[Function]}
        placeholder="Search a city..."
        type="text"
        value=""
      />
      <p
        className="message"
      >
        0
        Â Items found...
      </p>
    </form>
  </div>,
  <div
    className="cities-container"
  />,
]
`;
